src/ДЗ 14.1. Typescript. Основи ООП.ts

```diff
@@ -0,0 +1,160 @@
+// ======================
+// Declaration
+// ======================
+
+// 1 =====================================================================================================================
+
+interface INote {
+  name: string;
+  content: string;
+  createdAt?: Date;
+  editedAt?: Date;
+  status?: 'active' | 'done';
+}
+
+interface IConfirmationNote extends INote {
+  isEditable: boolean;
+}
+type NoteType = INote | IConfirmationNote;
+
+class Note<T extends NoteType> {
+  public notes: T[];
+
+  public get allNotes(): T[] {
```

Comment:
Why do you need all these accessors here? It seems like unnecessary functionality.

Reply:
You can implement all these accessors as ordinary methods instead.